### DISCUSSION TITLE
Question to UA4MT specification 8.3.1 Monitoring Element

### DISCUSSION THREAD

**Discussion by PeterSchreiner** [umati/UA-for-MachineTools#30] [Team Posts]
Dear Umati members
I have a specific question about the Umati specification at node 8.3.1 Monitoring Type
The node has a 0: HasComponent reference to an Object <MonitoringElement>. Is this notation with <>, following the rules of the OOP, to be understood as a list of monitored parameters?
More precisely: I want to publish a list of machine parameters for external monitoring (temperatures, speeds, axis states). Is that the right place?
Thank you in advance for your help

**Comment by GoetzGoerisch** [comment]
Dear @PeterSchreiner,

the `<_MontioredElement_>' is a placeholder for _Subtypes_ of the abstract [_ElementMonitoringType_](https://reference.opcfoundation.org/v104/MachineTool/v100/docs/8.3.2/) .
Therefore, only such subtypes are allowed.

Currently the _MonitoringTypes_ are by design such simple elements, because for the base specification this was not feasible to go already more detailed.

As you are speaking of axis and speeds etc. these are part of the CNC specification OPC 40502 which is currently being updated to new and changed requirements and can then be used in context of the 40501-1 to have a standardized model for such requirements.
Therefore, I ask you, if you have concrete needs please describe/contribute them to the subworking group to be included in the upcoming version?

CC: @TonjaHeinemann @umati/machinetool

**Comment by PeterSchreiner** [comment]
Thank you for your answer.
So far I've only focused on the OPC 40501 MachineTools. My object tree shows the nodes of the 40501 that are marked as mandatory.
Am I right in assuming that if I want to implement the 40502 I must / can build a second object tree parallel to the existing one?
Does the specification provide for an interface between the two trees?
As far as I can see, the 40502 maps large parts of the PLC.
